docs/mapping-annotations-for-AreaC.md

```diff
@@ -0,0 +1,200 @@
+# Practical Guide to Mapping Annotations for Parser dev
+
+This doc explains how to populate the NOMAD simulations datamodel using mapping annotations. It shows what they are, how to write them, and how to handle nested and repeating sections. 
+
+---
+
+## What a mapping annotation is
+
+- A small note on a schema field that says: *for source `<KEY>`, read value at `<PATH>` and put it here.*
+- Stored on any section/quantity: `m_annotations['mapping'][<KEY>] = MapperAnnotation(mapper=...)`.
+- `<KEY>` names the source (e.g., `OUT_KEY`, `HDF5_KEY`, `INFO_KEY`, `BAND_KEY`).
+- Paths use JMESPath style. Relative paths start with `.` (from the parent’s mapped node). `@` anchors at the source root.
+- The machinery (`MetainfoParser`) walks the schema, follows these notes, instantiates sections/subsections, and fills values.
+
+## What happens under the hood
+
+1) A source parser (`XMLParser`, `HDF5Parser`, `TextParser`, etc.) reads a file into a Python dict.
+2) Set `annotation_key` on a `MetainfoParser` to the `<KEY>` that matches the annotations.
+3) Call `convert(source_parser, target_parser)`. One key → one `convert`. Multiple keys/sources → run `convert` once per key after switching `annotation_key` each time.
+4) The target archive is populated; repeating subsections are created automatically from lists.
+
+## Advantages of using mapping
+
+- Declarative and readable: data movement is described alongside the schema, not scattered in code.
+- Reusable across sources: same schema, different annotation keys; swap sources by switching `annotation_key`.
+- Automatic instantiation: nested and repeating subsections are created based on list structure; no manual wiring.
+- Composable transforms: reshape, filter, and unit application via small helper functions in the parser.
```

Comment:
Agreed. Since you mentioned declarative style above, and these applications are (conceptually) immutable, you could even tentatively call it "functional" here. Jives really well with composition too.